language/tools/move-cli/src/base/movey_login.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use move_command_line_common::{env::MOVE_HOME, movey_constants::MOVEY_URL};
+use std::{fs, fs::File, io, path::PathBuf};
+use toml_edit::easy::{map::Map, Value};
+
+pub const MOVEY_CREDENTIAL_PATH: &str = "/movey_credential.toml";
+
+#[derive(Parser)]
+#[clap(name = "movey-login")]
+pub struct MoveyLogin;
+
+impl MoveyLogin {
+    pub fn execute(self) -> Result<()> {
+        println!(
+            "Please paste the API Token found on {}/settings/tokens below",
+            MOVEY_URL
+        );
+        let mut line = String::new();
+        loop {
+            match io::stdin().read_line(&mut line) {
+                Ok(_) => {
+                    line = line.trim().to_string();
+                    if !line.is_empty() {
+                        break;
+                    }
+                    println!("Invalid API Token. Try again!");
+                }
+                Err(err) => {
+                    bail!("Error reading file: {}", err);
+                }
+            }
+        }
+        Self::save_credential(line, MOVE_HOME.clone())?;
+        println!("Token for Movey saved.");
+        Ok(())
+    }
+
+    pub fn save_credential(token: String, move_home: String) -> Result<()> {
+        fs::create_dir_all(&move_home)?;
+        let credential_path = move_home + MOVEY_CREDENTIAL_PATH;
+        let credential_file = PathBuf::from(&credential_path);
+        if !credential_file.exists() {
+            let credential_file = File::create(&credential_path)?;
+            set_permissions(&credential_file, 0o600)?;
+        }
+
+        let old_contents: String;
+        match fs::read_to_string(&credential_path) {
+            Ok(contents) => {
+                old_contents = contents;
+            }
+            Err(error) => bail!("Error reading input: {}", error),
+        }
+        let mut toml: Value = old_contents
+            .parse()
+            .map_err(|e| anyhow::Error::from(e).context("could not parse input as TOML"))?;
```

Comment:
It would probably be useful to include file path in error message so that the user can act on it. For similar reason, you may add something like "Delete file to regenerate it with default settings".